edx/edx-enterprise
# [ENT-464] Skip audit track data passback for enrollments by flag.

**Description:** This PR allows audit track enrollment reporting (including data pass-back via SuccessFactors) to be configurable on a per-customer basis.

**JIRA:** [ENT-464](https://openedx.atlassian.net/browse/ENT-464)

**Dependencies:** None

**Merge deadline:** End of this week or July 31st

**Testing instructions:**

- The SAP test instance is basically that which is referenced at the beginning of https://openedx.atlassian.net/wiki/pages/viewpage.action?spaceKey=SOL&title=SAP+Testing+and+Configuration+notes
- The business sandbox has courses that I *believe* are not yet *exported* to the SAP test instance
- Once the courses are *exported* and then *imported* to the SAP instance, the SAP *admin* has to make them available for SAP *learners*.
- Once the courses are available for SAP *learners*, we can take the course through SAP (they have a UI for selecting these imported courses and taking them).
- Now we do the real "testing". We enroll in audit mode, finish the course.
- We enroll (possibly for another SAP test learner; there are many available) in verified mode.
- We make sure the enterprise customer has audit data reporting disabled.
- We run the learner data export command and, as far as I know, we will be able to view this data on the SAP test instance somehow (where exactly it'd appear is what I'll have to figure out still), if our logic was implemented correctly in the PR.
- We run the script again and see that nothing has changed because it won't send the same data twice.
- We now turn on audit data reporting for the customer and run the learner data export command again. This time, it will send audit data.
- We check to see if it was sent. If our PR's logic is good, it will work.

**Merge checklist:**

  **DO NOT** just add dependencies to `requirements/*.txt` files.

**Post merge:**

**Author concerns:** There was a lot of code-reading ramp-up I had to do here, so I _might_ have missed a spot. But then again, the tests build my confidence, so I think everything's OK.
